applications/satellite-imagery-geotiff.ipynb

```diff
@@ -0,0 +1,479 @@
+    "\n",
+    "First we define convenience functions to load the rescaling factors and transform a dataset. The red band is band 4 and near infrared is band 5."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def load_scale_factors(filename, band_number):\n",
+    "    with open(filename) as f:\n",
+    "        metadata = json.load(f)\n",
+    "    M_p = metadata['L1_METADATA_FILE'] \\\n",
+    "                  ['RADIOMETRIC_RESCALING'] \\\n",
+    "                  ['REFLECTANCE_MULT_BAND_{}'.format(band_number)]\n",
+    "    A_p = metadata['L1_METADATA_FILE'] \\\n",
+    "                  ['RADIOMETRIC_RESCALING'] \\\n",
+    "                  ['REFLECTANCE_ADD_BAND_{}'.format(band_number)]\n",
+    "    return M_p, A_p"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def calculate_reflectance(ds, band_number, metafile='meta.json'):\n",
+    "    M_p, A_p = load_scale_factors(metafile, band_number)\n",
+    "    toa = M_p * ds + A_p\n",
+    "    return toa"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "red = calculate_reflectance(red, band_number=4)\n",
+    "nir = calculate_reflectance(nir, band_number=5)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Because the transformation is composed of arithmetic operations, execution is delayed and the operations are parallelized automatically."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "dask.array<add, shape=(1, 7841, 7721), dtype=float64, chunksize=(1, 1024, 1024)>\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(red.variable.data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The resulting image has floating point data with magnitudes appropriate to reflectance."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "<xarray.DataArray (band: 1)>\n",
+      "array([1.2107])\n",
+      "Coordinates:\n",
+      "  * band     (band) int64 1\n",
+      "<xarray.DataArray (band: 1)>\n",
+      "array([-0.1])\n",
+      "Coordinates:\n",
+      "  * band     (band) int64 1\n",
+      "<xarray.DataArray (band: 1)>\n",
+      "array([0.059213])\n",
+      "Coordinates:\n",
+      "  * band     (band) int64 1\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(red.max(dim=['x', 'y']).compute())\n",
+    "print(red.min(dim=['x', 'y']).compute())\n",
+    "print(red.mean(dim=['x', 'y']).compute())"
```

Comment:
Could we do all these operations into only one aggregation with several operations?

Reply:
I've gone ahead and done this.